frame/democracy/src/lib.rs

```diff
@@ -1119,18 +1121,24 @@ decl_module! {
 		/// see `weight_for::note_preimage`
 		/// # </weight>
 		#[weight = weight_for::note_imminent_preimage::<T>((encoded_proposal.len() as u32).into())]
-		fn note_imminent_preimage(origin, encoded_proposal: Vec<u8>) {
+		fn note_imminent_preimage(origin, encoded_proposal: Vec<u8>) -> DispatchResultWithPostInfo {
 			Self::note_imminent_preimage_inner(ensure_signed(origin)?, encoded_proposal)?;
+			// We check that this preimage was not uploaded before in `note_imminent_preimage_inner`,
+			// thus this call can only be successful once. If successful, user does not pay a fee.
```

Comment:
>  thus this call can only be successful once.

Because I am not familiar with this pallet I request a test case for this statement, but ignore if it is trivial to prove.